presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -198,11 +221,35 @@ public HivePageSink(
             }
             else {
                 dataColumnsInputIndex.add(inputIndex);
+                dataColumnNameToIdMap.put(column.getName(), inputIndex);
+                dataColumnNameToTypeMap.put(column.getName(), column.getHiveType());
             }
         }
         this.partitionColumnsInputIndex = Ints.toArray(partitionColumns.build());
         this.dataColumnInputIndex = Ints.toArray(dataColumnsInputIndex.build());
 
+        requireNonNull(bucketProperty, "bucketProperty is null");
+        if (bucketProperty.isPresent()) {
+            int bucketCount = bucketProperty.get().getBucketCount();
+            checkArgument(bucketCount < MAX_BUCKET_COUNT, "bucketCount must be smaller than 100000");
```

Comment:
There isn't a particular reason why there must not be more buckets than 100,000 for reading.